### DISCUSSION TITLE
Amethyst 0.16.0 Beta

### DISCUSSION THREAD

**Discussion by ianyh** [ianyh/Amethyst#1265] [Announcements]
* Amethyst now more robustly supports configuration files. See documentation: [Configuration Files](https://github.com/ianyh/Amethyst/blob/development/docs/configuration-files.md) #1251
* Amethyst now supports a beta of custom layouts. See documentation: [Custom Layouts](https://github.com/ianyh/Amethyst/blob/development/docs/custom-layouts.md) #1134, #51, #1126 
* Fix the presence of ghost windows in the bsp layout. Thanks, @LudwigWS! #1249, #1246
* Fix the layout HUD when multiple screen are arranged vertically. Thanks, @Henkru! #1162, #1163
* Make TwoPane layout adaptive to screen orientation. Thanks, @mwz! #1182
* Improve main window swapping with TwoPane layout. Thanks, @rickythefox! #1241, #1242, #1234 
* Add 4Column layout for ultra-wide screens. Thanks, @reyk! #1154 
* Throw window to any space across multiple screens. Thanks, @dozzman! #1209 
* Fix an issue that caused Amethyst to fail to tile entirely. #1250 

<hr /><em>This discussion was created from the release <a href='https://github.com/ianyh/Amethyst/releases/tag/v0.16.0b'>Amethyst 0.16.0 Beta</a>.</em>

**Comment by mwz** [comment]
Thanks for the release @ianyh - would you be able to put it out in the dev channel?

**Reply by ianyh** [reply]
Hoping to distribute it today or tomorrow if I can find time.

**Comment by ianyh** [comment]
Finally out on the dev channel!

**Comment by freebrowser1** [comment]
The delay for focus follows mouse is still missing, making it useless as moving the mouse to another window pops up the underlying window over which the mouse moves to the intended target window.

**Reply by ianyh** [reply]
Okay? Snide, demanding comments like this are why open source maintainers burn out. What was this comment supposed to accomplish? My available time and energy for this "useless" software are both already extremely scarce. If you want a feature go thumbs up the Issue for it instead of killing my motivation to do _anything_. I try to make time for Amethyst because I care about the community, but I'm a single human being with a life, and I can only do so much.

**Reply by illettante** [reply]
> The chief joy of free and open-source software is that if a problem with a piece of software irritates one enough, there are no barriers of proprietary copyright enforcement preventing one from simply upskilling in Swift and fixing the problem oneself, thus bathing one in the threefold glow of the Contributor; the satisfaction of a problem solved by one's own ingenuity, the feeling of earning the respect of one's peers and gratitude of end-users, and the knowledge that one's improvement makes a valuable contribution and keeps a community of like-minded individuals thriving and healthy.

tl;dr thanks @ianyh, love your work

**Comment by freebrowser1** [comment]
Sorry but I am not demanding anything. I just tell that I still miss the delay. That's all.

**Comment by DitchComfort** [comment]
Just wondering if this release is relatively stable?

And, I know there’s a sample config file, however, I would like a sample file with every option in it.

**Reply by ianyh** [reply]
Yeah. There's one issue with setting commands from config file not working in most cases that I probably want to fix before full release, but I don't think there's anything functionally breaking.

I don't know when I'll have a chance to make a more robust sample config, though.

**Reply by DitchComfort** [reply]
Okey I just installed the latest beta. But how am I supposed to do stuff? I was trying to only change the modifiers from my config file relying on the sample config, but didn’t really do anything. I only have the "defaults" in the app itself. I got a message saying stuff could conflict because a config file was found. So for now I just deleted the config file and going back to use the settings from the app.

**Reply by DitchComfort** [reply]
Yeah I got most settings to work via the config file. I’m so excited for this and much more in tune with other apps/dotfiles 👍🏻 I think I spend some time setting up my file properly (incl. every command) so when stuff is ready am good to go.

**Comment by DitchComfort** [comment]
I did put together a more comprehensive sample config, but at the moment you get an error if you load the commands. But the general settings should be all fine.

[amethyst-sample.zip](https://github.com/ianyh/Amethyst/files/9083399/amethyst-sample.zip)

**Comment by pgroce** [comment]
Do you have any thoughts on when this version may come out of prerelease? I'm really looking forward to the BSP bugfix, but the config file and custom layout look like much larger changes that may take time to feel confident about. OTOH, it's also been a few months, so maybe you're almost there? 

If you feel like there's a lot of work left between now and 0.16.0 stable, might you consider backporting some of the smaller bugfixes to the 0.15.* series? (Again, personally I'm just waiting for the BSP fix. It's a lovely layout when it works, but I get enough ghost windows for it to be unusable without frequent Amethyst restarts.)

If you're lacking the spoons, I understand. I'd volunteer to do the backport myself, but I have neither Swift experience nor a dev toolchain installed for this project. Thanks for your work on this project.

**Reply by ianyh** [reply]
I was vaguely holding out for a bug with setting commands via config, but I think it should just go out.
